File: scripts/glm.R


###################
#### Libraries ####
###################

library(dplyr)

#######################
#### Main Function ####
#######################

run_glm <- function(){
  
  ###################
  #### Load Data ####
  ###################
  
  mod_dat <- data.table::fread("/Users/atlan/Adv_computing/AdvComp_final_proj/data/sim_data.csv",
                               colClasses = "numeric",
                               showProgress = T) 

  mod <- glm(y ~ ., data = mod_dat, 
             family = "binomial")
  
  return(mod)
}

glm_res <- run_glm()

time <- microbenchmark::microbenchmark(run_glm(),
                                       unit = "s",
                                       times = 10)

elapsed <- tibble(summary(time)) %>% 
  dplyr::select(-expr, -neval) %>% 
  mutate(across(everything(), ~./60)) #convert to minutes

out <- list(result_table = summary(glm_res)$coefficients,
            comp_time = elapsed)

save(out,
     file = "/Users/atlan/Adv_computing/AdvComp_final_proj/data/glm.Rdata")

File: scripts/Julia/glm.jl
using DataFrames, GLM, CSV, BenchmarkTools

function run_glm()
    mod_dat = CSV.read("/Users/atlan/Adv_computing/AdvComp_final_proj/data/sim_data.csv",
                   DataFrame, header = 1)
                   
    fm = @formula(y ~ X1 + X2 + X3 + X4)

    logit = glm(fm, mod_dat, Binomial(), LogitLink())
    temp = GLM.coeftable(logit)

    out = DataFrame(variable = temp.rownms,
                    Estimate = temp.cols[1],
                    StdError = temp.cols[2],
                    z_val = temp.cols[3],
                    lowerCI = temp.cols[5],
                    upperCI = temp.cols[6])
    return out
end

run_glm()
run_time = @elapsed res = run_glm()
run_time = round(run_time, digits = 4)
CSV.write("/Users/atlan/Adv_computing/AdvComp_final_proj/data/glm_$run_time.csv", res)